build/common.sh

```diff
@@ -1154,6 +1192,88 @@ function kube::release::gcs::verify_release_gt() {
   fi
 }
 
+# Check if the new version is greater than or equal to the version currently
+# published on GCS.  (Ignore the build; if it's different, overwrite anyway.)
+#
+# Globals:
+#   KUBE_GCS_PUBLISH_VERSION
+#   KUBE_GCS_RELEASE_BUCKET
+# Arguments:
+#   publish_file: the GCS location to look in
+# Returns:
+#   If new version is greater than the GCS version
+function kube::release::gcs::verify_ci_ge() {
+  local -r publish_file="${1-}"
+  local -r new_version=${KUBE_GCS_PUBLISH_VERSION}
+  local -r publish_file_dst="gs://${KUBE_GCS_RELEASE_BUCKET}/${publish_file}"
+
+  kube::release::parse_and_validate_ci_version "${new_version}" || return 1
+
+  local -r version_major="${BASH_REMATCH[1]}"
+  local -r version_minor="${BASH_REMATCH[2]}"
+  local -r version_patch="${BASH_REMATCH[3]}"
+  local -r version_prerelease="${BASH_REMATCH[4]}"
+  local -r version_alpha_rev="${BASH_REMATCH[5]}"
+  local -r version_commits="${BASH_REMATCH[7]}"
+
+  local gcs_version
+  if gcs_version="$(gsutil cat "${publish_file_dst}")"; then
+    kube::release::parse_and_validate_ci_version "${gcs_version}" || {
+      kube::log::error "${publish_file_dst} contains invalid ci version, can't compare: '${gcs_version}'"
+      return 1
+    }
+
+    local -r gcs_version_major="${BASH_REMATCH[1]}"
+    local -r gcs_version_minor="${BASH_REMATCH[2]}"
+    local -r gcs_version_patch="${BASH_REMATCH[3]}"
+    local -r gcs_version_prerelease="${BASH_REMATCH[4]}"
+    local -r gcs_version_alpha_rev="${BASH_REMATCH[5]}"
+    local -r gcs_version_commits="${BASH_REMATCH[7]}"
+
+    local greater=true
+    if [[ "${version_major}" -lt "${gcs_version_major}" ]]; then
+      greater=false
+    elif [[ "${version_major}" -gt "${gcs_version_major}" ]]; then
+      : # fall out
+    elif [[ "${version_minor}" -lt "${gcs_version_minor}" ]]; then
+      greater=false
+    elif [[ "${version_minor}" -gt "${gcs_version_minor}" ]]; then
+      : # fall out
+    elif [[ "${version_patch}" -lt "${gcs_version_patch}" ]]; then
+      greater=false
+    elif [[ "${version_patch}" -gt "${gcs_version_patch}" ]]; then
+      : # fall out
+    elif [[ "${version_prerelease}" =~ alpha.* && "${gcs_version_prerelease}" == "beta" ]]; then
```

Comment:
Okay, nevermind.  I can't figure out a way to capture "alpha", (as opposed to "alpha.9",) without making things *really* ugly, which means I don't think we can rely on lexicographic comparison.  (Because of the case above, that "alpha.9" < "alpha.90")